app/src/main/java/org/oppia/android/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -445,29 +436,32 @@ class StatePlayerRecyclerViewAssembler private constructor(
     val textView = checkNotNull(congratulationsTextView) {
       "Expected non-null reference to congratulations text view"
     }
-    checkNotNull(bannerConfettiView) {
+    val confettiView = checkNotNull(congratulationsTextConfettiView) {
       "Expected non-null reference to confetti view"
     }
+    val colorsList = checkNotNull(confettiColors) {
+      "Expected non-null list for confetti colors"
+    }
 
-    createBannerConfetti()
+    createBannerConfetti(confettiView, colorsList)
 
     textView.visibility = View.VISIBLE
 
     val fadeIn = AlphaAnimation(0f, 1f)
     fadeIn.interpolator = DecelerateInterpolator()
-    fadeIn.duration = 2000
+    fadeIn.duration = 1500
 
     val fadeOut = AlphaAnimation(1f, 0f)
     fadeOut.interpolator = AccelerateInterpolator()
-    fadeOut.startOffset = 1000
+    fadeOut.startOffset = 1500
     fadeOut.duration = 1000
 
     val animation = AnimationSet(false)
     animation.addAnimation(fadeIn)
     animation.addAnimation(fadeOut)
     textView.animation = animation
 
-    lifecycleSafeTimerFactory.createTimer(2000).observe(
+    lifecycleSafeTimerFactory.createTimer(2500).observe(
```

Comment:
For this & the animation times above/below, are these all related now? If so, we should use relatively computed constants defined at the top of the file to ensure they stay consistently correct.

Reply:
Done

Reply:
Much clearer--thanks!